types/proposal.go

```diff
@@ -42,6 +42,32 @@ func NewProposal(height int64, round int, blockPartsHeader PartSetHeader, polRou
 	}
 }
 
+// ValidateBasic performs basic validation.
+func (p *Proposal) ValidateBasic() error {
+	if p.Height < 0 {
+		return errors.New("Negative Height")
+	}
+	if p.Round < 0 {
+		return errors.New("Negative Round")
+	}
```

Comment:
timestamp?

Reply:
how should validate it? compare it to the local time with some delta?

Reply:
Yeh, maybe plus or minus a year? Just to eliminate the crazy outliers. The decoder will have done alot of work for us here already actually. We can punt on it for now.